src/backend/aspen/app/views/api_utils.py

```diff
@@ -35,3 +35,29 @@ def format_datetime(dt: Optional[datetime.datetime], format="%Y-%m-%d %I:%M%p")
         return dt.strftime(format)
     else:
         return "N/A"
+
+
+def check_data(
+    sample_fields: list[str],
+    pathogen_genome_fields: list[str],
+    required_fields: tuple,
+    optional_fields: tuple,
+) -> Tuple[bool, List[str], List[str]]:
+
+    combined_fields: list[str] = sample_fields + pathogen_genome_fields
+    # check all required fields are present:
+    missing_required: list[str] = [
+        i for i in required_fields if i not in combined_fields
+    ]
+
+    # check no fields were added that are unexpected
+    unexpected: list[str] = [
+        i for i in combined_fields if i not in required_fields + optional_fields
+    ]
+
+    if missing_required or unexpected:
+        return False, missing_required, unexpected
+
+    else:
```

Comment:
I don't think you need the else here!

Reply:
so my thinking was that i kind of liked the true being returned because then we can say` if data_ok` in the view otherwise we would have to say if `data_ok ==None` in the view? I can take it out though if it's confusing!

Reply:
oh yea keep the return! I just meant you didn't need the else. Could just be 
```
if missing_required or unexpected:
        return False, missing_required, unexpected
return True, [], []
```